include/swift/SIL/SILInstruction.h

```diff
@@ -1001,6 +1029,9 @@ class ApplyInstBase<Impl, Base, false> : public Base {
   /// The type of the callee with our substitutions applied.
   SILType SubstCalleeType;
 
+  /// Information about specialization and inlining of this apply.
```

Comment:
You should add a comment that this is only != null if the apply was inlined. And in this case it points to the specialization info of the inlined function.